strongbox-security/strongbox-authentication-registry/src/test/java/org/carlspring/strongbox/authentication/ExternalUserDetailsServiceTest.java

```diff
@@ -0,0 +1,128 @@
+package org.carlspring.strongbox.authentication;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.UUID;
+
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+
+import org.carlspring.strongbox.config.hazelcast.HazelcastConfiguration;
+import org.carlspring.strongbox.config.hazelcast.HazelcastInstanceId;
+import org.carlspring.strongbox.domain.UserEntry;
+import org.carlspring.strongbox.users.dto.User;
+import org.carlspring.strongbox.users.dto.UserDto;
+import org.carlspring.strongbox.users.service.UserService;
+import org.carlspring.strongbox.users.service.impl.EncodedPasswordUser;
+import org.carlspring.strongbox.users.service.impl.OrientDbUserService.OrientDb;
+import org.carlspring.strongbox.users.service.impl.YamlUserService.Yaml;
+import org.carlspring.strongbox.users.userdetails.StrongboxUserManager;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.Primary;
+import org.springframework.context.annotation.Profile;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.transaction.annotation.Transactional;
+
+@SpringBootTest
+@ActiveProfiles({ "test", "ExternalUserDetailsServiceTestConfig" })
+@TestPropertySource(properties = { "strongbox.config.file.authentication.providers=classpath:eudst-strongbox-authentication-providers.xml",
+                                   "strongbox.authentication.providers.yaml=classpath:/etc/conf/eudst-strongbox-authentication-providers.yaml",
+                                   "users.external.cache.seconds=3600",
+                                   "strongbox.users.config.yaml=classpath:/etc/conf/eudst-strongbox-security-users.yaml" })
+@ContextConfiguration(classes = TestConfig.class)
+public class ExternalUserDetailsServiceTest
```

Comment:
You don't have `ExternalUserDetailsService`

Reply:
yeah, this probably confusing as well by `ExternalUserDetailsService` I've mean the `ConfigurableProviderManager` which implements the `UserDetailsService` and used as default implementation in our Spring Security

Reply:
renamed to `ConfigurableProviderManagerTest` with implementation type assertion in test